cosmic-applet-power/src/main.rs

```diff
@@ -101,73 +130,139 @@ impl Application for Power {
+                            PowerAction::Restart => Command::perform(restart(), Message::Zbus),
+                            PowerAction::Shutdown => Command::perform(shutdown(), Message::Zbus),
+                        },
+                    ])
+                } else {
+                    Command::none()
+                }
+            }
+            Message::Cancel => {
+                if let Some((id, _)) = self.action_to_confirm.take() {
+                    return destroy_layer_surface(id);
+                }
+                Command::none()
+            }
+            Message::Closed(id) => {
+                if let Some((surface_id, _)) = self.action_to_confirm {
+                    if id == surface_id {
+                        self.action_to_confirm = None;
+                        return destroy_layer_surface(id);
+                    }
+                }
+                if id == window::Id(0) {
+                    process::exit(0);
+                }
+                Command::none()
+            }
+            Message::Timeout(id) => {
+                if let Some((surface_id, a)) = self.action_to_confirm {
+                    if id == surface_id {
+                        self.action_to_confirm = None;
+                        return Command::batch(vec![
+                            destroy_layer_surface(id),
+                            match a {
+                                PowerAction::Lock => Command::perform(lock(), Message::Zbus),
+                                PowerAction::LogOut => Command::perform(log_out(), Message::Zbus),
+                                PowerAction::Suspend => Command::perform(suspend(), Message::Zbus),
+                                PowerAction::Restart => Command::perform(restart(), Message::Zbus),
+                                PowerAction::Shutdown => {
+                                    Command::perform(shutdown(), Message::Zbus)
+                                }
+                            },
+                        ]);
+                    }
+                }
+                Command::none()
+            }
         }
     }
 
     fn view(&self, id: window::Id) -> Element<Message> {
-        if id == window::Id::new(0) {
-            self.applet_helper
-                .icon_button(&self.icon_name)
-                .on_press(Message::TogglePopup)
-                .into()
-        } else {
-            let settings = row_button(vec!["Settings...".into()]).on_press(Message::Settings);
+        if matches!(self.popup, Some(p) if p == id) {
+            let settings =
+                row_button(vec![text(fl!("settings")).size(14).into()]).on_press(Message::Settings);
 
             let session = column![
                 row_button(vec![
                     text_icon("system-lock-screen-symbolic", 24).into(),
-                    "Lock Screen".into(),
+                    text(fl!("lock-screen")).size(14).into(),
```

Comment:
Would we be interested in building "pallette" widgets for cosmic? Rather than having to `text(...).size(...)` all the time, we could just expose a few 'h1()' `h2()` `paragraph()` helper widgets in libcosmic?

(This is obviously not blocking, just an idea)